lib/propono/services/queue_listener.rb

```diff
@@ -9,18 +9,25 @@ def self.drain(*args, &message_processor)
       new(*args, &message_processor).drain
     end
 
-    attr_reader :aws_client, :propono_config, :topic_name, :visibility_timeout, :message_processor
+    attr_reader :aws_client, :propono_config, :topic_name, :visibility_timeout, :message_processor, :idle_timeout
     def initialize(aws_client, propono_config, topic_name, options = {}, &message_processor)
       @aws_client = aws_client
       @propono_config = propono_config
       @topic_name = topic_name
       @message_processor = message_processor
       @visibility_timeout = options[:visibility_timeout] || nil
+      @idle_timeout = options[:idle_timeout] || nil
     end
 
     def listen
       raise ProponoError.new("topic_name is nil") unless topic_name
+
+      reset_idle_timeout!
+
       loop do
+        if idle_timeout_reached?
```

Comment:
Mind changing to `break if idle_timeout_reached?`?

Reply:
Refactored in 029b968.